proposed/extended-coding-style-guide.md

```diff
@@ -38,6 +38,7 @@ This example encompasses some of the rules below as a quick overview:
 
 ~~~php
 <?php
+
```

Comment:
Declare usually goes w/o blank line as was noted in discussion at mailing list.

Reply:
In the discussion Woody mentions what is commonly found in existing code, not what we should aim for.

Why should we make an exceptional case for `declare` ? How is it simpler to say, sometimes we must write:

```php
<?php
declare(strict_types=1);

use namespace Foo;

$a = 123;
```

But then sometime we must write:

```php
<?php

/**
 * Some comment
 */

declare(strict_types=1);

use namespace Foo;

$a = 123;
```

Allowing no blank line before `declare` in **some** cases, make the rules harder to remember, harder to use, and look inconsistent across files of the same project.

Reply:
Making `declare` special makes little to no sense to me.

I MIGHT be convinced that "the first thing after the php tag" could omit the newline" is a worthwhile exception, but only if that's the case regardless of what that thing is.  Making an exception only sometimes just makes it harder for the human parser and machine parser.

Reply:
I also agree with @Crell here.

Reply:
Maybe people used to regard this line break as a break before the namespace (or any other code), not after the `<?php`. Or as a separation between regular PHP code and "redundant header stuff".

Since `<?php` and `declare(strict_types=1)` will usually be the same in all files across a package, people have seen it as part of the "opening shebang", which needs to be separated from the rest.

Not having a line break before `declare()` makes it easier to visually skip this part and focus on the things that are specific to this file.

Reply:
@donquixote, the "opening shebang", as you call it, would most likely be the PHP opening tag immediately followed by a file-level docblock in many open source projects, because it is common practice to include some boilerplate about license. A `declare` statement would only come after those.

If you are willing to add boilerplate headers of several lines (which again seems to be a common practice and brings absolutely no value in terms of pure coding), and somehow be able mentally "skip" those, you might as well tolerate a single more blank line.